packages/thirdweb/src/wallets/coinbase/coinbaseSDKWallet.ts

```diff
@@ -334,11 +334,21 @@ function createAccount(provider: ProviderInterface, address: string) {
         params: [account.address, stringifiedData],
       });
     },
+    onTransactionRequested: async () => {
+      // make sure to show the coinbase popup BEFORE doing any transaction preprocessing
+      // otherwise the popup might get blocked in safari
+      await showCoinbasePopup(provider);
+    },
   };
 
   return account;
 }
 
+export async function showCoinbasePopup(provider: ProviderInterface) {
+  // biome-ignore lint/suspicious/noExplicitAny: based on the latest CB SDK - scary but works
+  await (provider as any).communicator?.waitForPopupLoaded();
```

Comment:
we should ask coinbase to add this to their spec officially so it gets covered by some sort of semver

Reply:
also are there versions where the coinbase provider does *not* have this function?

Reply:
basically, if we're adding this we need to carry the burden of understanding it fully for now

Reply:
yep im thinking of putting up a PR on their repo directly.

also we have a pinned version so we should be ok for now